src/redux/openIdConfigSlice.ts

```diff
@@ -28,6 +31,14 @@ export const fetchOpenIdConfiguration = createAsyncThunk<OpenIdConfigPayload, st
     }
 );
 
+export const fetchOpenIdConfigurationIfNecessary = (openIdConfigUrl: string):
+    ThunkAction<void, RootState, unknown, AnyAction> =>
+    async (dispatch, getState) => {
+        const { isFetching, data, expiry } = getState().openIdConfiguration;
+        if (isFetching || data || (expiry && Date.now() < expiry * 1000)) return;
```

Comment:
cache doesn't work anymore here, since it will always return if `data` is truthy. remove that from the `or` clause here